test/integration/dialects/abstract/connection-manager.test.ts

```diff
@@ -0,0 +1,186 @@
+import { getTestDialect, getTestDialectTeaser, createSequelizeInstance } from '../../support';
+
+const expect = chai.expect;
+const baseConf = Config[getTestDialect()];
+const poolEntry = {
+  host: baseConf.host,
+  port: baseConf.port,
+  pool: {},
+};
+
+const dialect = getTestDialect();
+
+describe(getTestDialectTeaser('Connection Manager'), () => {
+  let sandbox: SinonSandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.createSandbox();
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('does not initialize a readOnly pool without replication', () => {
+    const options = {
+      replication: null,
+    };
+
+    const sequelize = createSequelizeInstance(options);
+    expect(sequelize.connectionManager.pool).to.be.instanceOf(ReplicationPool);
+    expect(sequelize.connectionManager.pool.read).to.be.null;
+    expect(sequelize.connectionManager.pool.write).to.be.instanceOf(Pool);
+  });
+
+  it('initializes a readOnly pool with replication', () => {
+    const options = {
+      replication: {
+        write: { ...poolEntry },
+        read: [{ ...poolEntry }, { ...poolEntry }],
+      },
+    };
+
+    const sequelize = createSequelizeInstance(options);
+    expect(sequelize.connectionManager.pool).to.be.instanceOf(ReplicationPool);
+    expect(sequelize.connectionManager.pool.read).to.be.instanceOf(Pool);
+    expect(sequelize.connectionManager.pool.write).to.be.instanceOf(Pool);
+  });
+
+  it('should round robin calls to the read pool', async () => {
+    if (getTestDialect() === 'sqlite') {
+      return;
+    }
+
+    const slave1 = { ...poolEntry };
+    const slave2 = { ...poolEntry };
+    slave1.host = 'slave1';
+    slave2.host = 'slave2';
+
+    const options = {
+      replication: {
+        write: { ...poolEntry },
+        read: [slave1, slave2],
+      },
+    };
+
+    const sequelize = createSequelizeInstance(options);
+    const connectionManager = sequelize.connectionManager;
+
+    const res: Connection = {};
+
+    const connectStub = sandbox.stub(connectionManager, '_connect').resolves(res);
+    sandbox.stub(connectionManager, '_disconnect').resolves();
+    sandbox.stub(sequelize, 'databaseVersion').resolves(sequelize.dialect.defaultVersion);
+
+    const queryOptions: GetConnectionOptions = {
+      type: 'read',
+      useMaster: false,
+    };
+
+    const _getConnection = connectionManager.getConnection.bind(
+      connectionManager,
+      queryOptions,
+    );
+
+    await _getConnection();
+    await _getConnection();
+    await _getConnection();
+    chai.expect(connectStub.callCount).to.equal(4);
+
+    // First call is the get connection for DB versions - ignore
+    const calls = connectStub.getCalls();
+    chai.expect(calls[1].args[0].host).to.eql('slave1');
+    chai.expect(calls[2].args[0].host).to.eql('slave2');
+    chai.expect(calls[3].args[0].host).to.eql('slave1');
+  });
+
+  // sqlite does not have different database versions since it's bundled
```

Comment:
Would still be nice to check, because maybe an older version of the connector might work but have an older database version. Or do we have a separate test for the minimum connector version?

Reply:
Let's keep it for another PR. It's not a bug introduced by this PR, but this PR uncovered it. This test was broken and always ran against AbstractConnectionManager instead of SQLiteConnectionManager - so this wasn't caught